src/main/java/org/apache/maven/model/jdom/JDomDependencies.java

```diff
@@ -123,53 +121,26 @@ public Dependency set(int index, Dependency dependency) {
 
   @Override
   public void add(int index, Dependency dependency) {
-    addInternal(dependency, index);
-  }
+    if (index > size() || index < 0) {
+      throw new IndexOutOfBoundsException("Index: " + index + ", Size: " + size());
+    }
+
+    int elementIndex = index;
+    if (index > 0) {
+      Element previousElement = ((JDomDependency) get(index - 1)).getJDomElement();
+      elementIndex = 1 + getElementIndex(previousElement, jdomElement);
```

Comment:
IMHO it is very confusing to import JDomUtils statically. I suggest to use `JDomUtils.xxxx` instead

Reply:
For me, either way is OK. Code is shorter with static imports, but I understand that readability might be better using `JDomUtils.xxxx `. Let's discuss with @rreich and @fwienber